### PR TITLE
Upgraded cairo to 2.7.0 & Implement OP_SHA256

### PR DESCRIPTION
**Pull request by supreme2580**
- [x] issue # https://github.com/keep-starknet-strange/shinigami/issues/114
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/shinigami/blob/main/CONTRIBUTING.md)
- [x] code change includes tests

Upgraded to cairo 2.7.0 and implemented OP_SHA256

### CODE REVIEW

**Comment by supreme2580** [comment]
@b-j-roberts it may seem like I touched multiple files, that only happened because I ran ```scarb fmt``` before pushing and it formatted other files

**Comment by b-j-roberts** [comment]
Looks good, I'll merge ASAP.
Would you be interested in implementing `OP_SHA256` ontop of this PR? You can start on that whenever if so.

I'd recommend an online tool like this to get sample test data https://emn178.github.io/online-tools/sha256.html

**Comment by supreme2580** [comment]
@b-j-roberts I will implement ```OP_SHA256``` ontop this PR, thank you

**Comment by supreme2580** [comment]
Hi @b-j-roberts I have just implemented ``` OP_SHA256 ```

File: src/opcodes/tests/test_stack.cairo [hunk 1]
```diff
@@ -269,3 +269,12 @@ fn test_op_2rot_insufficient_items() {
     let mut engine = utils::test_compile_and_run_err(program, Error::STACK_OUT_OF_RANGE);
     utils::check_dstack_size(ref engine, 5);
 }
+
+#[test]
+fn test_opcode_sha256() {
+    let program = "OP_1 OP_SHA256";
+    let mut engine = utils::test_compile_and_run(program);
+    utils::check_dstack_size(ref engine, 1);
+    let expected_dstack = array!["1274352175"];
```
Comment (b-j-roberts) [inline src/opcodes/tests/test_stack.cairo:278]: I believe the resulting stack should be `array![hex_to_bytecode("0x4bf5122f344554c53bde2ebb8cd2b7e3d1600ad631c385a5d7cce23c7785459a")]`.
If you use this ide https://ide.scriptwiz.app, you can change the witness script to `OP_1 OP_SHA256` to see and get other examples too.

**Review by b-j-roberts — CHANGES_REQUESTED** [review]

File: src/opcodes/tests/test_stack.cairo (hunk 1 above)
Comment (supreme2580) [reply src/opcodes/tests/test_stack.cairo:278]: I see... working on it

**Review by b-j-roberts — APPROVED** [review]
Awesome! First hash function down 👍

**Pull request merged**